src/main/java/uk/gov/hmcts/darts/authentication/config/AuthenticationConfiguration.java

```diff
@@ -14,4 +19,15 @@ public AuthConfigFallback getNoAuthConfigurationFallback(DefaultAuthConfiguratio
             throw new DartsApiException(AuthenticationError.FAILED_TO_OBTAIN_AUTHENTICATION_CONFIG);
         };
     }
+
+    @Bean
+    public AuditorAware<UserAccountEntity> auditorAware(UserIdentity userIdentity) {
```

Comment:
How does this behave if we try to save an entity with an explicitly set `createdBy`/`lastModifiedBy` field? Does the value get replaced with the result of `userIdentity.getUserAccount()`, or is the original value respected?

Asking because I came across a scenario where we set them explicitly, rather than picking from `userIdentity` in `TransformedMediaHelper.createTransformedMediaEntity()`.

Reply:
This is a very good point, and that is done for good reason so we don't really want that to be overridden.

Reply:
Not sure about other usages sorry. I think you may have to look at all usages of `.setCreatedBy()` and `.setLastModifiedBy()` on `UserAccountEntity` to check if any other code is manually setting the values, and address the impact there, unless you can think of a smarter way.

Another example that I found using that approach is `UserManagementServiceImpl.createUser()`, which I think will arrive at the same `userIdentity.getUserAccount()` value, but I suppose we'd want to eliminate the manual setting so that we're not setting the same value twice?

Reply:
I have updated the way we auto update createdBy and lastModifiedBy to use EntityLiseners to allow us to more easily bypass the auto update logic such that all existing logic should now be unaffected by this change